packages/common/src/api/tan-query/utils/useLineupQuery.ts

```diff
@@ -62,11 +70,44 @@ export const useLineupQuery = ({
     lineup.status
   ])
 
+  const [lineupTrackIds, lineupCollectionIds] = useMemo(() => {
+    const [tracks, collections] = partition(
+      lineup.entries,
+      (entry) => entry.kind === Kind.TRACKS
+    )
+    return [
+      tracks.map((entry) => entry.id),
+      collections.map((entry) => entry.id)
+    ]
+  }, [lineup.entries])
+
+  const { data: tracks } = useTracks(lineupTrackIds)
+  const { data: collections } = useCollections(lineupCollectionIds)
+
+  const entries: LineupEntry<LineupTrack | Track | Collection>[] =
+    useMemo(() => {
+      return lineup.entries.map((entry) => {
+        const entity =
+          entry.kind === Kind.TRACKS
+            ? tracks?.find((track) => track.track_id === entry.id)
+            : entry.kind === Kind.COLLECTIONS
+              ? collections?.find(
+                  (collection) => collection.playlist_id === entry.id
+                )
+              : entry
+        return {
+          ...entry,
+          ...entity
+        } as LineupEntry<LineupTrack | Track | Collection>
+      })
+    }, [lineup.entries, tracks, collections])
+
```

Comment:
neat.
The mapping and mapping and mapping is a little gnarly but I don't know that this could be much cleaner given how lineups work.

Reply:
could be a useEntities hook that can handle arbitrary entities? but that may also be even more gnarly? im trying to think outside of lineup if there are other cases where we have a list of different types of entities...